### PR TITLE
[BE] Fix MYPY issues

### PR DESCRIPTION
**Pull request by aorenste**
Fix some mypy issues that have crept in to the trunk.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133872

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @ezyang @malfet @xuzhao9 @gramster @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by oulgen — APPROVED** [review]

File: torch/utils/_ordered_set.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 
 # Using Generic[T] bc py38 does not support type parameterized MutableSet
-class OrderedSet(Generic[T], MutableSet):
+class OrderedSet(MutableSet, Generic[T]):
```
Comment (Skylion007) [inline torch/utils/_ordered_set.py:26]: Weird that ruff didn't catch this?

File: torch/utils/_sympy/value_ranges.py [hunk 2]
```diff
@@ -97,11 +98,11 @@ def sympy_generic_le(lower, upper):
         return not (lower and not upper)
 
 
-def vr_is_bool(vr: ValueRanges[_T]) -> TypeIs[ValueRanges[SympyBoolean]]:
+def vr_is_bool(vr: ValueRanges[_T]) -> TypeGuard[ValueRanges[SympyBoolean]]:
```
Comment (Skylion007) [inline torch/utils/_sympy/value_ranges.py:101]: No, these should be TypeIs imported from typing extensions

Comment (Skylion007) [reply torch/utils/_sympy/value_ranges.py:101]: What are the errors they are causing out of curisosity?

File: torch/utils/_sympy/value_ranges.py [hunk 3]
```diff
@@ -97,11 +98,11 @@ def sympy_generic_le(lower, upper):
...
     return vr.is_bool
 
 
-def vr_is_expr(vr: ValueRanges[_T]) -> TypeIs[ValueRanges[sympy.Expr]]:
+def vr_is_expr(vr: ValueRanges[_T]) -> TypeGuard[ValueRanges[sympy.Expr]]:
```
Comment (Skylion007) [inline torch/utils/_sympy/value_ranges.py:105]: I guess should be TypeGuard and maybe the other value should be TypeIs?

File: torch/utils/_sympy/value_ranges.py (hunk 2 above)
Comment (aorenste) [reply torch/utils/_sympy/value_ranges.py:101]: The problem is that TypeIs requires the resulting type to be a strict narrowing of the original type.  This is not so it's getting an [narrowed-type-not-subtype] error.

**Review by Skylion007 — APPROVED** [review]

File: torch/utils/_sympy/value_ranges.py (hunk 3 above)
Comment (Skylion007) [reply torch/utils/_sympy/value_ranges.py:105]: TypeIs is an unused import now, and ruff should complain

File: torch/utils/_sympy/value_ranges.py [hunk 4]
```diff
@@ -14,10 +14,10 @@
     overload,
     SupportsFloat,
     TYPE_CHECKING,
+    TypeGuard,
```
Comment (Skylion007) [inline torch/utils/_sympy/value_ranges.py:17]: needs to be from typing_extensions since we still support 3.9

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**